cap/one_for_one.go

```diff
@@ -42,40 +38,38 @@ func oneForOneRestartLoop(
 	supCtx context.Context,
 	eventNotifier EventNotifier,
 	supRuntimeName string,
+	supTolerance *errToleranceManager,
 	supChildren map[string]c.Child,
 	supNotifyCh chan<- c.ChildNotification,
-	wasComplete bool,
 	prevCh c.Child,
 	prevErr error,
-) *c.ErrorToleranceReached {
+) *ErrorToleranceReached {
 	for {
+		if prevErr != nil {
+			ok := supTolerance.checkTolerance()
```

Comment:
I can't exactly put my finger on it, but it feels like the `prevErr` should be encapsulated in the tolerance?

Reply:
Check the comment on the `prevErr` declaration above